server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -443,6 +443,13 @@ public ClusterState applyCreateIndexRequest(
 
             // The backing index may have a different name or prefix than the data stream name.
             final String name = request.dataStreamName() != null ? request.dataStreamName() : request.index();
+            // Do not apply any templates to system indices
+            // Using applyCreateIndexRequestWithV1Templates with empty list instead of applyCreateIndexRequestWithV2Template with null
+            // template as applyCreateIndexRequestWithV2Template has assertions when template is null
+            if (systemIndices.isSystemIndex(name)) {
+                return applyCreateIndexRequestWithV1Templates(currentState, request, silent, Collections.emptyList(), metadataTransformer);
```

Comment:
I like this approach but would prefer you refactor this to keep consistent naming, basically, create a new method `applyCreateIndexRequestWithNoTemplates()` that just delegates to this V1 templates method with the empty list argument.   Then in the conditional here, call that other method.